### PR TITLE
nrf_security: Remove random_ext.c

### PR DESCRIPTION
**Pull request by Vge0rge**
Remove the random_ext.c file and the MBEDTLS_PSA_CRYPTO_EXTERNAL_RNG option since it is not used anymore by the Oberon PSA core.

With the current configuration the options:
PSA_CRYPTO_DRIVER_ALG_CTR_DRBG_*

are used to specifiy the random number generator algorithm.

Ref: NCSDK-21868

Signed-off-by: Georgios Vasilakis <georgios.vasilakis@nordicsemi.no>

### CODE REVIEW

**Review by vili-nordic — APPROVED** [review]
lgtm

File: subsys/nrf_security/Kconfig.legacy [hunk 1]
```diff
@@ -345,19 +345,9 @@ config MBEDTLS_HMAC_DRBG_C
...
 
-config MBEDTLS_PSA_CRYPTO_EXTERNAL_RNG
-	bool
-	prompt "External RNG"
-	default y if CRYPTOCELL_USABLE
-	help
-	  Convenience layer for handling entropy before it is fully supported in
-	  PSA Crypto APIs and drivers. This is set by default for legacy and PSA
-	  modes.
-
```
Comment (SebastianBoe) [inline subsys/nrf_security/Kconfig.legacy:356]: I see this has a prompt and we have a patch to openthread. Is it possible for a user to be selecting or enabling this?

You might want to consider increasing the test coverage by temporarily touching west.yml.

**Review by SebastianBoe — APPROVED** [review]

File: subsys/nrf_security/configs/psa_crypto_config.h.template [hunk 2]
```diff
@@ -236,7 +236,6 @@
 /* MBEDTLS_PSA_CRYPTO_DRIVERS is defined to 1 by TF-M's build system. */
 #cmakedefine MBEDTLS_PSA_CRYPTO_DRIVERS                         @MBEDTLS_PSA_CRYPTO_DRIVERS@
 #cmakedefine MBEDTLS_PSA_CRYPTO_CLIENT
-#cmakedefine MBEDTLS_PSA_CRYPTO_EXTERNAL_RNG
```
Comment (joerchan) [inline subsys/nrf_security/configs/psa_crypto_config.h.template:239]: `#define MBEDTLS_PSA_CRYPTO_EXTERNAL_RNG`

Needed to pass check-config rule.

File: subsys/nrf_security/Kconfig.legacy (hunk 1 above)
Comment (joerchan) [reply subsys/nrf_security/Kconfig.legacy:356]: Not needed I think. Openthread is using mbedtls entropy, not PSA entropy.

File: tests/crypto/src/common_test.h [hunk 3]
```diff
@@ -56,7 +56,7 @@ size_t hex2bin_safe(const char *hex, uint8_t *buf, size_t buflen);
 #include "psa/crypto.h"
 #include "nrf_cc3xx_platform_ctr_drbg.h"
 
-extern mbedtls_psa_external_random_context_t drbg_ctx;
+extern char drbg_ctx;
```
Comment (joerchan) [inline tests/crypto/src/common_test.h:59]: This declaration does not match well with the #elif of common_tests.c.

**Review by joerchan — APPROVED** [review]

**Review by torsteingrindvik — APPROVED** [review]

**Pull request merged**